src/wallet/wallet.cpp

```diff
@@ -1155,6 +1155,30 @@ void CWallet::BlockConnected(const std::shared_ptr<const CBlock>& pblock, const
     for (size_t i = 0; i < pblock->vtx.size(); i++) {
         SyncTransaction(pblock->vtx[i], pindex, i);
     }
+
+
+    if (pindex) {
+        // Only notify UI if this transaction is in this wallet
+        if (hashPrevBestCoinbase.IsNull()) {
+            // For correctness we scan over the entire wallet, looking for
+            // the previous block's coinbase, just in case it is ours, so
+            // that we can notify the UI that it should now be displayed.
```

Comment:
I expanded a bit on the comment. The commit message is correct, if its hard to read, any suggestions for improvements?

I removed the outer-level if statement, as it is now useless, indeed.